### PR TITLE
Tickets/sitcom 426

### PR DESCRIPTION
**Pull request by isotuela**
Hi Patrick! I hope you're well. If you could please take a look at the SITCOM-426 PSF&Photometry notebook, I'd really appreciate it. Cheers!
Ioana

### CODE REVIEW

File: notebooks/summit/AuxTel/SITCOM-426/SITCOM-426_PSFandPhotometry_Validation.ipynb [hunk 1]
```diff
@@ -0,0 +1,918 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bd4c8b32-d03c-48b2-a2cf-61df3b9ad6b1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# exposure = await latiss.take_engtest(\n",
```
Comment (tribeiro) [inline notebooks/summit/AuxTel/SITCOM-426/SITCOM-426_PSFandPhotometry_Validation.ipynb]: I know this is commented out, but, when taking acquisition images, please use `latiss.take_acq`, so the images have the correct header info.

File: notebooks/summit/AuxTel/SITCOM-426/SITCOM-426_PSFandPhotometry_Validation.ipynb [hunk 2]
```diff
@@ -0,0 +1,918 @@
...
+    "\n",
+    "steps = np.arange(z_offset_start, z_offset_end + 0.01, z_offset_step)\n",
+    "parabolic_focus_data = []\n",
+    "\n",
+    "await atcs.rem.ataos.cmd_offset.set_start(z=z_offset_start)\n",
+    "    \n",
+    "await asyncio.sleep(2)\n",
+    "for step in steps:\n",
+    "        \n",
+    "    exposure = await latiss.take_engtest(\n",
```
Comment (tribeiro) [inline notebooks/summit/AuxTel/SITCOM-426/SITCOM-426_PSFandPhotometry_Validation.ipynb]: When doing focus sweep, please, use `latiss.take_focus`.

File: notebooks/summit/AuxTel/SITCOM-426/SITCOM-426_PSFandPhotometry_Validation.ipynb [hunk 3]
```diff
@@ -0,0 +1,918 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e62e70c2-b45d-4a96-98e0-eedba84b038d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# 2 sec exposures\n",
+    "await latiss.take_engtest(\n",
```
Comment (tribeiro) [inline notebooks/summit/AuxTel/SITCOM-426/SITCOM-426_PSFandPhotometry_Validation.ipynb]: I think the next exposures should all be `take_object` no?

**Review by tribeiro — COMMENTED** [review]
I spoke with @patrickingraham  and I think we can execute this test entirely with script. Seems like Chris is ok with doing CWFS instead of the classical focus sweep. 

In any case, I left a few comments you may want to consider for future reference.

Also, if you agree in executing this test with scripts. I recommend taking a look at the new queue classes:

```
from lsst.ts.observatory.control.auxtel import ATQueue

queue = ATQueue(domain)
```

There's a series of new method do help validate script configuration and add scripts to the script queue.

File: notebooks/summit/AuxTel/SITCOM-426/SITCOM-426_SQ_PSFandPhotometry_Validation.ipynb [hunk 4]
```diff
@@ -0,0 +1,1190 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "abd66b8a-ed7f-4f0d-ba02-e5211bf375f9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "async def send_to_SQ(is_standard, script_path, script_config):\n",
```
Comment (tribeiro) [inline notebooks/summit/AuxTel/SITCOM-426/SITCOM-426_SQ_PSFandPhotometry_Validation.ipynb]: Is this really necessary? If you don't want to use `add_standard` and `add_external`, you can use `add` instead:

```
await queue.add(
    is_standard=take_cwfs_is_standard,
    ....
)
```

**Review by tribeiro — CHANGES_REQUESTED** [review]
Instead of defining the `send_to_SQ` method, please, use `queue.add`, which does the exact same thing. 

Also, one thing that worries me is that the notebook is laid out such that you do:

```
script 1:
    define configuration
    validate configuration
    add to queue
script 2:
    define configuration
    validate configuration
    add to queue
...
```

The problem with this approach is that if the validation fails for one script, you already added the previous ones to the queue. 

I think it would be better if you did something like

```
Define configuration for script1
Define configuration for script2
Define configuration for script3
....

Validate config for script1
Validate config for script2
Validate config for script3
...

Add script1 to queue
Add script2 to queue
Add script2 to queue
....
```

I know this might be a lot or refactoring to do now but something to consider the next time maybe?

File: notebooks/summit/AuxTel/SITCOM-426/SITCOM-426_SQ_PSFandPhotometry_Validation.ipynb (hunk 4 above)
Comment (isotuela) [reply notebooks/summit/AuxTel/SITCOM-426/SITCOM-426_SQ_PSFandPhotometry_Validation.ipynb]: Thanks! This is great. I didn't see this `add` method when I was writing this :)

**Comment by isotuela** [comment]
Thank you for your comments. 

While I agree and the notebook originally was written as you suggested, I decided, when running it at TTS, to group each script (Configuration, validation and add to queue) together. The reason is simplicity of use, in case we need to modify the configuration, run certain script again, etc. I also don't see an issue in this case to have any of these scripts queued in the SQ. It was too cumbersome going back to the beginning of the notebook to change exposure time for a certain script or the coordinates of the source for CWFS.

**Pull request merged**